src/glmfit.jl

```diff
@@ -269,6 +308,41 @@ function loglikelihood(m::AbstractGLM)
     ll
 end
 
+function nullloglikelihood(m::GeneralizedLinearModel{<:GlmResp{<:Any,<:Any,L}}) where L
+    hasintercept(m) ||
+        throw(ArgumentError("model does not have an intercept, " *
+                            "null log-likelihood cannot be interpreted"))
+
+    r      = m.rr
+    wts    = r.wts
+    y      = r.y
+    d      = r.d
+    offset = r.offset
+    ll  = zero(eltype(y))
+    if isempty(r.offset) # Faster method
+        if !isempty(wts)
+            mu = mean(r.y, weights(wts))
+            ϕ = nulldeviance(m)/sum(wts)
+            @inbounds for i in eachindex(y, wts)
+                ll += loglik_obs(d, y[i], mu, wts[i], ϕ)
+            end
+        else
+            mu = mean(r.y)
+            ϕ = nulldeviance(m)/length(y)
+            @inbounds for i in eachindex(y)
+                ll += loglik_obs(d, y[i], mu, 1, ϕ)
+            end
+        end
+    else
+        nullm = fit(GeneralizedLinearModel,
+                    fill(1.0, length(y), 1), y, d, L(), wts=wts, offset=offset,
+                    maxiter=m.maxiter, minstepfac=m.minstepfac,
+                    atol=m.atol, rtol=m.rtol)
```

Comment:
I'm not sure whether this is really needed, as the null model should converge much more quickly than more complex models. Maybe we don't need to use a low tolerance just to compute the deviance.

Note that this requires storing these values in new `GeneralizedLinearModel` fields (which could be useful in its own right).